internal/controller/nonadminbackup_controller_test.go

```diff
@@ -465,69 +491,94 @@ var _ = ginkgo.Describe("Test full reconcile loop of NonAdminBackup Controller",
 				gomega.Expect(err).ToNot(gomega.HaveOccurred(), "failed to run manager")
 			}()
 			// wait manager start
-			time.Sleep(1 * time.Second)
+			managerStartTimeout := 10 * time.Second
+			pollInterval := 100 * time.Millisecond
+			ctxTimeout, cancel := context.WithTimeout(ctx, managerStartTimeout)
+			defer cancel()
+
+			err = wait.PollUntilContextTimeout(ctxTimeout, pollInterval, managerStartTimeout, true, func(ctx context.Context) (done bool, err error) {
+				select {
+				case <-ctx.Done():
+					return false, ctx.Err()
+				default:
+					// Check if the manager has started by verifying if the client is initialized
+					return k8sManager.GetClient() != nil, nil
+				}
+			})
+			// Check if the context timeout or another error occurred
+			gomega.Expect(err).ToNot(gomega.HaveOccurred(), "Manager failed to start within the timeout period")
 
 			ginkgo.By("Waiting Reconcile of create event")
-			nonAdminBackup := buildTestNonAdminBackup(nonAdminNamespaceName, scenario.spec)
-			gomega.Expect(k8sClient.Create(ctx, nonAdminBackup)).To(gomega.Succeed())
+			nonAdminBackup := buildTestNonAdminBackup(nonAdminObjectNamespace, nonAdminObjectName, scenario.spec)
+			gomega.Expect(k8sClient.Create(ctxTimeout, nonAdminBackup)).To(gomega.Succeed())
```

Comment:
is not better to use `ctxTimeout` only in `PollUntilContextTimeout`? my fear is if the test takes more than 10 seconds, it will break

Reply:
The initial implementation had only `time.Sleep()` for `1 second`. It was static sleep expecting the manager will start and be ready after that period of time.

I've changed this to be pulled ever `100 ms` but with limit of `10 seconds`, which is 10x more what it was before.

Assuming that something will be ready after 1 second was not ideal that's why I've added that retry logic here.